demos/applications/apa102.cpp

```diff
@@ -0,0 +1,148 @@
+// Copyright 2024 Khalil Estell
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <libhal-display/apa102.hpp>
+#include <libhal-util/serial.hpp>
+#include <libhal-util/steady_clock.hpp>
+
+#include "../resource_list.hpp"
+
+hal::byte build_brightness_byte(int p_brightness)
```

Comment:
```suggestion
hal::byte build_brightness_byte(int p_brightness)
```

Still need to make this unsigned.